src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        stateCopy = {};
+        break;
+
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+        break;
+
+      default:
+        return 'Error';
```

Comment:
[CODE STYLE]: A `default` case in a `switch` statement should handle unexpected cases without stopping the function's execution. Instead of returning an error directly, consider throwing an exception or continuing the loop with a warning. This allows the function to handle all actions in the input array and return a complete history of states.

Reply:
Fix this comment